problems/[230608] Number of Islands.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * leetcode problem link: https://leetcode.com/problems/number-of-islands
+ *
+ * @param {character[][]} grid
+ * @return {number}
+ */
+const numIslands = function (grid) {
+  let islandCount = 0;
+
+  for (let row = 0; row < grid.length; row++) {
+    for (let column = 0; column < grid[row].length; column++) {
+      if (grid[row][column] === "1") {
+        islandCount += findCliff(row, column);
+      }
+    }
+  }
+
+  return islandCount;
+
+  function findCliff(row, column) {
```

Comment:
```suggestion
  function findCliff(grid, row, column) {
```

지금은 grid 변수의 참조를 위해서 내부 함수로 동작하지만 함수를 외부로 분리할 수도 있을 것 같습니다~ 아마 순수해지려면 grid를 복사한 배열을 변경시키기고 반환해야겠지만요. 성능 상의 이점은 줄어들겠네요..